### ISSUE TITLE
Lower torch.roll

### ISSUE THREAD

**Issue by rwightman** [pytorch/xla#3427]
## 🚀 Feature
A lowering of torch.roll would be helpful so code model code using it can be common across gpu/tpu. 

```
pt-xla-profiler: Op(s) not lowered: aten::roll,  Please open a GitHub issue with the above op lowering requests.
pt-xla-profiler: TransferFromServerTime too frequent: 20 counts during 1 steps
```

## Additional context

Roll can be implemented fairly easily with cat across relevant dimensions, however, torch.roll can be specified across multiple dim, which are done in the same CUDA kernel so making code common by reverting to cat will slow down models on GPU.

**Comment by JackCaoG** [comment]
Thanks Ross, we will take a look.

**Comment by cicirori** [comment]
@rwightman 

I implemented a version of roll's lower mainly refer to [tf2xla roll op](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/tf2xla/kernels/roll_op.cc), and in the meantime I read [torch roll_cuda implementation]( https://github.com/pytorch/pytorch/blob/344ecb2e7103b4d92ec3859318643a66dd50a4ed/aten/src/ATen/native/cuda/TensorTransformations.cu#L102)

Unfortunately with multiple dimensional rolls, torch also seems to run a separate cuda kernel for each shift/dim pair? 

I haven't analyzed the tf2xla implementation in detail, but it looks like it's not very efficient in terms of memory usage, I'm not sure.

**Comment by JackCaoG** [comment]
roll lowering is merged, closing this issue.
